subprojects/docs/src/docs/userguide/working_with_files.adoc

```diff
@@ -173,7 +173,13 @@ include::sample[dir="userguide/files/archives/groovy",files="build.gradle[tags=u
 include::sample[dir="userguide/files/archives/kotlin",files="build.gradle.kts[tags=unpack-archive-example]"]
 ====
 
-As with a normal copy, you can control which files are unpacked via filters and even rename files as they are unpacked.
+As with a normal copy, you can control which files are unpacked via filters and even rename files as they are unpacked. Sometimes you'll only need to unpack a subset of the archive. In that case, you can use api:org.gradle.api.tasks.AbstractCopyTask#eachFile(org.gradle.api.Action)[] to apply an action to each file as it is about to be copied into its destination. You can change the relative path of the files to the another root, like so:
```

Comment:
Would you mind forward linking "filters" to the `#filtering_files` anchor and "rename files" to the `#sec:renaming_files` anchor? I know that text is already there, but I just noticed that readers would benefit from the links.

Reply:
> Sometimes you'll only need to unpack a subset of the archive

Can't you just use `include()` and `exclude()` filters to do that? As I understand it, this is supposed to document the specific example of extracting different subtrees to different paths/directories within the destination.

I think you should start a new paragraph here because we want to present this as a separate example. We also need to be much clearer in _what_ is being demonstrated. Consider something like:

> More advanced processing can be handled by the api:org.gradle.api.tasks.AbstractCopyTask#eachFile(org.gradle.api.Action)[eachFile()] method. For example, you might need to extract different subtrees of the archive into different paths within the destination directory. The following sample uses the method to extract the files within the archive's `libs` directory into the root destination directory, rather than into `$destDir/libs`:

I hope that's correct. Even if it's not, I hope the point is clear. We need to help the reader understand what we're showing them.

Reply:
With the `include()` and `exclude()` comment, I was referring specifically to the comment

> Sometimes you'll only need to unpack a subset of the archive

If you just want to unpack a subset without changing the root, you can do that with `include()` and `exclude()` settings, right?

As for empty directories, that should be mentioned. I would use a `[CAUTION]` block:

```
[CAUTION]
====
You can not change the destination path of empty directories
with this technique. You can learn more in <url-to-issue>[this issue].
====
```

Then make sure that the corresponding issue mentions that this part of the docs needs updating when it's fixed/implemented.

Reply:
Done. A note was added to issue https://github.com/gradle/gradle/issues/2940